### PR TITLE
Replace getSavedData() with getFavoritesForRecord().

### PR DESCRIPTION
**Pull request by demiankatz**
This backports and adapts the work from https://github.com/demiankatz/vufind/pull/28 -- the remaining work from there will be incorporated into #2233 when this is merged.

### CODE REVIEW

File: module/VuFind/src/VuFind/AjaxHandler/GetSaveStatuses.php [hunk 1]
```diff
@@ -56,28 +57,34 @@ class GetSaveStatuses extends AbstractBase implements TranslatorAwareInterface
...
-     * Format list object into array.
+     * Format UserResourceEntityInterface object into array.
      *
-     * @param array $list List data
+     * @param UserResourceEntityInterface $data UserResourceEntityInterface object
      *
      * @return array
      */
-    protected function formatListData($list)
+    protected function formatListData(UserResourceEntityInterface $data)
```
Comment (aleksip) [inline module/VuFind/src/VuFind/AjaxHandler/GetSaveStatuses.php]: `: array`

File: module/VuFind/src/VuFind/AjaxHandler/GetSaveStatuses.php [hunk 2]
```diff
@@ -56,28 +57,34 @@ class GetSaveStatuses extends AbstractBase implements TranslatorAwareInterface
...
     {
+        $list = $data->getUserList();
```
Comment (aleksip) [inline module/VuFind/src/VuFind/AjaxHandler/GetSaveStatuses.php:83]: `null` is a legal return value for`getUserList()`, do we need to take that into account here?

File: module/VuFind/src/VuFind/Controller/AbstractRecord.php [hunk 3]
```diff
@@ -492,13 +493,14 @@ public function saveAction()
...
+        $resources = $this->getDbService(UserResourceServiceInterface::class)->getFavoritesForRecord(
             $driver->getUniqueId(),
+            $driver->getSourceIdentifier(),
             null,
-            $driver->getSourceIdentifier()
+            $user
         );
         foreach ($resources as $userResource) {
-            $listIds[] = $userResource->list_id;
+            $listIds[] = $userResource->getUserList()->getId();
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Controller/AbstractRecord.php]: `null` is a legal return value for `getUserList()`, do we need to take that into account here?

File: module/VuFind/src/VuFind/Controller/MyResearchController.php [hunk 4]
```diff
@@ -1010,14 +1011,16 @@ public function editAction()
         }
 
         // Get saved favorites for selected list (or all lists if $listID is null)
-        $userResources = $user->getSavedData($id, $listID, $source);
+        $userResourceService = $this->getDbService(UserResourceServiceInterface::class);
+        $userResources = $userResourceService->getFavoritesForRecord($id, $source, $listID, $user);
         $savedData = [];
         foreach ($userResources as $current) {
+            $currentList = $current->getUserList();
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Controller/MyResearchController.php]: `null` is a legal return value for `getUserList()`, do we need to take that into account here?

File: module/VuFind/src/VuFind/Controller/MyResearchController.php [hunk 5]
```diff
@@ -1026,10 +1029,10 @@ public function editAction()
         // to a particular list ID:
         $containingLists = [];
         if (!empty($listID)) {
-            $userResources = $user->getSavedData($id, null, $source);
+            $userResources = $userResourceService->getFavoritesForRecord($id, $source, null, $user);
         }
         foreach ($userResources as $current) {
-            $containingLists[] = $current->list_id;
+            $containingLists[] = $current->getUserList()->getId();
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Controller/MyResearchController.php]: `null` is a legal return value for `getUserList()`, do we need to take that into account here?

**Review by aleksip — APPROVED** [review]

**Comment by demiankatz** [comment]
Thanks, @aleksip! I've addressed your feedback in commit 2fb2620ecdb71cb23fcbf12c99921b72a4413738, and tests are still passing, so I'm merging now.

**Pull request merged**